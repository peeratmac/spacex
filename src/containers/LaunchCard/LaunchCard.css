.launch-card {
  padding: 1rem;
  border: #ccc 1px solid;
  margin: 0.7rem 0;
}

.launch-card-main {
  color: black;
}

.site-name {
  min-height: 50px;
}

.save-button {
  display: inline-block;
  background: lightblue;
  border-radius: 10px;
  padding: 0.4rem 1.3rem;
  font-size: 1rem;
  border: none;
  cursor: pointer;
  margin-right: 0.5rem;
  outline: none;
}
