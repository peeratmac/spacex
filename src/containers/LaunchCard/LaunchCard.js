import React from 'react';
import { NavLink } from 'react-router-dom';
import './LaunchCard.css';

export const LaunchCard = ({ launch, saveFavorite }) => {
  return (
    <NavLink
      className='launch-card-main'
      to={`/launch/${launch.flight_number}`}
      style={{ textDecoration: 'none' }}>
      <div
        onClick={() => console.log(`hi, ${launch.links.mission_patch}`)}
        className='launch-card'>
        <p>Flight Number: {launch.flight_number}</p>
        <p>Mission: {launch.mission_name}</p>
        <p>Launch: {launch.launch_year}</p>
        <p className='site-name'>Site: {launch.launch_site.site_name_long}</p>
        <button onClick={() => saveFavorite(launch)} className='save-button'>
          SAVE
        </button>
      </div>
    </NavLink>
  );
};

export default LaunchCard;
