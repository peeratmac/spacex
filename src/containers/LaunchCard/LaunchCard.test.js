import React from 'react';
import { shallow } from 'enzyme';
import { LaunchCard } from './LaunchCard';

describe('LAUNCH CARD', () => {
  it.skip('should match the LaunchCard Snapshot UI', () => {
    let wrapper = shallow(<LaunchCard launch={[]} saveFavorite={jest.fn()} />);

    expect(wrapper).toMatchSnapshot();
  });
});
