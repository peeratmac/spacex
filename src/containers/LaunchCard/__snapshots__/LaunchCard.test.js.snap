// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LAUNCH CARD should match the LaunchCard Snapshot UI 1`] = `ShallowWrapper {}`;
